pkg/alertmanager/amcfg.go

```diff
@@ -99,24 +99,50 @@ func (c alertmanagerConfig) String() string {
 	return string(b)
 }
 
+type enforcer interface {
+	processRoute(*route) *route
+	processInhibitRule(*inhibitRule) *inhibitRule
+	updateNamespace(namespace string)
+}
```

Comment:
I'm not sure that we need updateNamespace() here. To keep it generic maybe the 2 other functions need to have an additional parameter holding name+namespace of the referenced AlertmanagerConfig object?


```suggestion
type enforcer interface {
	processRoute(types.NamespacedName, *route) *route
	processInhibitRule(types.NamespacedName, *inhibitRule) *inhibitRule
}
```